scripts/test.py

```diff
@@ -197,10 +220,45 @@ def render(args):
         with open(output_file, 'r') as file:
             if file.read() != rendered:
                 fail('circle CI configuration is out of date, re-render it')
-    else:
+    elif not args.required:
         with open(output_file, 'w') as file:
             file.write(rendered)
 
+    # now for mergify!
+    mergify_conf_dir = os.path.dirname(args.mergify_jinja_file)
+    jinja_file_basename = os.path.basename(args.mergify_jinja_file)
+    (output_file, ext) = os.path.splitext(args.mergify_jinja_file)
+    assert ext == '.jinja'
+
+    test_permutations = filter_test_permutations(['*'], required_blacklist, permutations=test_permutations)
+
+    env = jinja2.Environment(loader=jinja2.FileSystemLoader(mergify_conf_dir), autoescape=False)
+    template = env.get_template(jinja_file_basename)
+
+    required_status = list(filter(lambda el: el not in not_required_status_checks,
+            chain(
+            ("ci/circleci: %s" % job for job in
+                chain(("test--%s" % profile for profile in test_permutations.keys()),
+                      ("test-unit--%s" % profile for profile in unit_test_profiles),
+                      ("build-artifacts--%s" % profile for profile in build_artifact_profiles))),
+            extra_required_status_checks
+             )))
+
+    rendered = template.render(
+        required_status=required_status
+    )
+
+    if args.check:
+        with open(output_file, 'r') as file:
+            if file.read() != rendered:
+                fail('mergify configuration is out of date, re-render it')
+    elif not args.required:
```

Comment:
It's not entirely clear to me why there is this special case logic for required. Couldn't you just pass in `--check` to not write the files and then `--required` will only print out the extra info you want? Or you could even put `required` as it's own top level command.

Reply:
With the current logic, `--required` will always print out the required status checks. If we used `--check`, the script might fail with no effect if somehow master got a stale `.mergify.yml`/`.circleci/config.yml`.

Reply:
It should just be a top-level command.